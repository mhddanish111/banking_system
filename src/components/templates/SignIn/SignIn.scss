@import '../../../styles/boarder.scss';
@import '../../../styles/spacing.scss';
@import '../../../styles/mixins.scss';
.sign-in {
    .main-container {
        display: flex;
        justify-content: center;
    }
}
.sign-in-container {
    border: $border-medium;
    border-radius: $border-radius;
    box-shadow: $box-shadow-focus;
    height: 500px;
    width: 40%;
    margin-top: 5%;
    @include mobile {
        width: 80%;
    }
    @include tablet {
        width: 60%;
    }
    .error-message-container{
        margin: 0 ($space-l - $space-xxs);
    }
}

.sign-in-heading {
    display: flex;
    justify-content:flex-start;
    margin: $space-xxs;
    padding: $space-xxs;
}