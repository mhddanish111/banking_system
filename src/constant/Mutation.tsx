import { gql } from '@apollo/client';

export const CREATE_USER = gql`
  mutation CreateUser($userInput: UserInput) {
    createUser(userInput: $userInput) {
      firstName
      lastName
      message
    }
  }
`;

export const CHANGE_PASSWORD = gql`
  mutation ChangePassword($changePasswordInput: ChangePasswordInput) {
    changePassword(changePasswordInput: $changePasswordInput) {
      message
      success
      data {
        firstName
      }
    }
  }
`;

export const OPEN_ACCOUNT = gql`
  mutation CreateAccount($accountInput: AccountInput) {
    createAccount(accountInput: $accountInput) {
      _id
      accountNumber
      amount
      accountType
      categoryName
      limit
      minAmount
      maxAmount
      accountTypeId
      categoryId
      userId
      message
      type
      debitCardNumber
      debitCardCSVNumber
      expiryYear
      expiryMonth
    }
  }
`;

export const DEPOSIT = gql`
  mutation Deposit($depositInput: DepositInput) {
    deposit(depositInput: $depositInput) {
      transactionResult {
        _id
      }
      accounts {
        _id
        type
        accountType
        categoryName
        accountNumber
        amount
        minAmount
        maxAmount
        limit
        charge
        debitCardNumber
        debitCardCSVNumber
        expiryYear
        expiryMonth
        users {
          _id
          firstName
          lastName
        }
      }
    }
  }
`;

export const WITHDRAWAL = gql`
  mutation Withdrawal($withdrawalInput: WithdrawalInput) {
    withdrawal(withdrawalInput: $withdrawalInput) {
      transactionResult {
        _id
      }
      accounts {
        _id
        type
        accountType
        categoryName
        accountNumber
        amount
        minAmount
        maxAmount
        limit
        charge
        debitCardNumber
        debitCardCSVNumber
        expiryYear
        expiryMonth
        users {
          _id
          firstName
          lastName
        }
      }
    }
  }
`;
