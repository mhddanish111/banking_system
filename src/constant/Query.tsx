import { gql } from '@apollo/client';

export const LOGIN_USER = gql`
  query Login($logInInput: LogInInput) {
    login(logInInput: $logInInput) {
      firstName
      lastName
      userId
      token
      isValid
      message
      userType
    }
  }
`;

export const USER_DETAILS = gql`
  query UserDetails {
    userDetails {
      _id
      firstName
      lastName
      email
      mobile
      occupation
      income
      aadhaarNumber
      panNumber
      govtId
      govtIdNumber
      address
      city
      state
      postalCode
      isActive
      message
    }
  }
`;

export const GET_ACCOUNT_TYPES = gql`
  query AccountTypes {
    accountType {
      _id
      name
      type
      isActive
      category {
        _id
        name
        benefit
        isActive
        minAmount
        maxAmount
        limit
        charge
      }
    }
  }
`;

export const GET_ACCOUNTS = gql`
  query Accounts {
    accounts {
      _id
      type
      accountType
      categoryName
      accountNumber
      amount
      minAmount
      maxAmount
      limit
      charge
      debitCardNumber
      debitCardCSVNumber
      expiryYear
      expiryMonth
      users {
        _id
        firstName
        lastName
      }
    }
  }
`;

export const GET_ACCOUNT_WITH_TYPE = gql`
  query AccountWithType($accountWithTypeInput: AccountWithTypeInput) {
    accountWithType(accountWithTypeInput: $accountWithTypeInput) {
      _id
      type
      accountType
      categoryName
      accountNumber
      amount
      minAmount
      maxAmount
      limit
      charge
      debitCardNumber
      debitCardCSVNumber
      expiryYear
      expiryMonth
      users {
        _id
        firstName
        lastName
        aadhaarNumber
        panNumber
      }
    }
  }
`;

export const GET_TRANSACTION = gql`
  query Transaction($transactionInput: TransactionInput) {
    transaction(transactionInput: $transactionInput) {
      count {
        offset
        limit
        totalCount
      }
      result {
        _id
        transactionId
        accountNumber
        amount
        withdrawalAmount
        depositAmount
        transactionDate
        accountType
        categoryName
        transactionType
      }
    }
  }
`;
