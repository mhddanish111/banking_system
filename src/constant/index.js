export const USER_KEY = 'user_key';

export const AccountTypeList = [
  { label: 'Accounts', value: 'Accounts' },
  { label: 'Cards', value: 'Cards' },
];
