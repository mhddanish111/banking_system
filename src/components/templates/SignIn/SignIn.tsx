import React, { useState } from 'react';
import { useLazyQuery } from '@apollo/client';
import { useNavigate } from 'react-router-dom';
import { addUser } from '../../../context/actions';
import Header from '../../organisms/Header';
import Heading from '../../atoms/Heading';
import SignInForm from '../../molecules/SignInForm';
import ErrorMessage from '../../atoms/ErrorMessage';
import { LOGIN_USER } from '../../../constant/Query';
import { setCookie } from '../../../utils';
import GetContext from '../../../context/GetContext';
import './SignIn.scss';

type formData = {
  user: string;
  password: string;
};

const SignIn: React.FC<any> = (): any => {
  const context = GetContext();
  const { dispatch }: any = context;
  const navigate = useNavigate();
  const [loginError, setLoginError] = useState<String>('');
  const [getLogin] = useLazyQuery(LOGIN_USER);
  const SignInHandler = (signInData: formData, e: any) => {
    e.preventDefault();
    const { user, password } = signInData;
    const logInInput = {
      email: user,
      password,
    };
    getLogin({ variables: { logInInput } })
      .then((res) => {
        if (res.data) {
          const {
            isValid,
            message,
            firstName,
            lastName,
            token,
            userType,
            userId,
          } = res.data?.login || {};
          if (!isValid) {
            setLoginError(message);
          } else {
            const data = {
              firstName,
              lastName,
              userType,
              userId,
            };
            dispatch(addUser(data));
            setCookie(token, 1);
            navigate('/dashboard');
          }
        }
      })
      .catch((error) => {
        if (error) {
          setLoginError(error.message);
        }
      });
  };
  const closeError = (e: any) => {
    e.preventDefault();
    setLoginError('');
  };
  return (
    <div className="sign-in">
      <Header isDefault={false} />
      <main className="main-container">
        <div className="sign-in-container">
          <Heading
            HeadingType="h1"
            className="sign-in-heading"
            ariaLabel="sign in title"
          >
            Sign In
          </Heading>
          {loginError && (
            <ErrorMessage
              message={loginError}
              className="alert"
              onClick={closeError}
            />
          )}
          <SignInForm SignInHandler={SignInHandler} />
        </div>
      </main>
    </div>
  );
};

export default SignIn;
