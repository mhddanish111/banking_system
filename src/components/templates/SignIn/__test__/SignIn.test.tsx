import '@testing-library/jest-dom';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SignIn from '../SignIn';
import { LOGIN_USER } from '../../../../constant/Query';

describe('<SignIn />', () => {
  const props = {
    selectedAccountNumber: 1673894730214,
    setIsCallTransaction: jest.fn(),
    isCallTransaction: true,
  };
  const logInInput = {
    email: 'test@test.com',
    password: '12345678',
  };
  const mocks = [
    {
      request: {
        query: LOGIN_USER,
        variables: {
          logInInput,
        },
      },
      result: {
        data: {
          login: {
            isValid: true,
            message: 'test',
            firstName: 'test',
            lastName: 'test',
            token: 'test test test test',
            userType: 'Admin',
            userId: 'user123',
          },
        },
      },
    },
  ];
  const mockedUsedNavigate = jest.fn();
  jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useNavigate: () => mockedUsedNavigate,
  }));
  const wrapper = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <BrowserRouter>
        <SignIn {...props} />
      </BrowserRouter>
    </MockedProvider>,
  ).toJSON;
  it('should render the SignIn Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
