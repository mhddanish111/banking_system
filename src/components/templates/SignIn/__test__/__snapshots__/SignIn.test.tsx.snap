// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should render the SignIn Component correctly 1`] = `[Function]`;
